### PR TITLE
Add GitHub library

### PR DESCRIPTION
**Pull request by radu-matei**
This PR finally introduces a GitHub library so we can encapsulate all GitHub logic in a single place.

It introduces a `Check` class that can be used as follows (actual code that generates tje checks for this PR):

```
function build(e, project) {
  var build = new Job(`${projectName}-build`, jsImg);

  build.tasks = [
    "cd /src",
    "yarn install",
    "yarn compile",
    "yarn test",
    "yarn audit"
  ];

  return build;
}

events.on("check_suite:requested", (e, p) => {
  var check = new Check(e, p, build());
  check.run();
});
```

This PR also checks in the compiled `out/` directory, so we can always test using the latest changes.

This only introduces the most basic functionality, and we should decide how much custom logic should go into the library vs. each project implementing their own - for example for releases. 

That being said, finally! 🎉

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
 node_modules/
-out/
+# out/
```
Comment (vdice) [inline .gitignore:4]: Should we just remove this line, then?

File: src/github.ts [hunk 2]
```diff
@@ -10,28 +38,35 @@ export class Notification {
...
     name: string;
     externalID: any;
-    detailsUrl: string;
+    detailsUrl: string = "";
     title: string;
     text: string;
     summary: string;
     count: number;
     conclusion: Conclusion;
+    notificationJobImage: string = "deis/brigade-github-check-run:latest";
```
Comment (vdice) [inline src/github.ts]: We should update this to `brigadecore/brigade-github-check-run` for the image name.  For tag, I'd vote for the latest tagged release, which is `v0.1.0` currently.  However, we'll have to remember to bump this tag each time a new release is out...

File: test/github.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import "mocha";
```
Comment (vdice) [inline test/github.ts:1]: nice tests!

**Review by vdice — COMMENTED** [review]
A few comments/suggestions.  Also, I see some files have been deleted (`acr.ts`, `o365.ts`, `slack.ts`) - is there a reason for this?  I'm assuming they could be refreshed to work similar to the added `github.ts` library, right?

File: .gitignore (hunk 1 above)
Comment (radu-matei) [reply .gitignore:4]: Rather than removing it, I'm adding a comment specifying why it's included.

**Comment by radu-matei** [comment]
@vdice - the rest of the files were deleted because they weren't working just yet.
We can revamp those as we start needing them (we can get them from the previous commit history).

**Review by vdice — APPROVED** [review]
LGTM!

**Pull request merged**